SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -168,144 +107,98 @@ @implementation OWSSendMessageOperation
 
 - (instancetype)initWithMessage:(TSOutgoingMessage *)message
                   messageSender:(OWSMessageSender *)messageSender
-                        success:(void (^)(void))aSuccessHandler
-                        failure:(void (^)(NSError *_Nonnull error))aFailureHandler
+                   dbConnection:(YapDatabaseConnection *)dbConnection
+                        success:(void (^)(void))successHandler
+                        failure:(void (^)(NSError *_Nonnull error))failureHandler
 {
     self = [super init];
     if (!self) {
         return self;
     }
 
-    _operationState = OWSSendMessageOperationStateNew;
-    self.backgroundTask = [OWSBackgroundTask backgroundTaskWithLabelStr:__PRETTY_FUNCTION__];
-
+    self.remainingRetries = 6;
```

Comment:
Note I bumped this up to 6. 

Previously the uploading process used the same retry counter as message sending.

As I recall it, we wanted a lower retry count because upload retries could take a *really* long time.

e.g. the scenario:

1. Alice sends Bob a large attachment
2. Alice sends Bob a short text message

Since sending the attachment (1) blocks sending the message (2), if Alice is on a bad network, it can take a really long time before Alice even tries to fire off (2).

In this case, we want (1) to fail relatively fast if it means that (2) can have a chance to succeed.

So: Uploads still retry 4 times, but message sends now retry a couple more times. Happy to tweak these, but my intuition is that the desired retry behavior for uploading attachments is not the same as that for message sending.

Reply:
Should we use a different value here based on the number of pre-send uploads?